application/src/main/resources/org/opentripplanner/apis/gtfs/schema.graphqls

```diff
@@ -4603,6 +4603,42 @@ input TransferPreferencesInput {
   slack: Duration
 }
 
+"A collection of selectors for what routes/agencies should be included in or excluded from the search."
+input TransitFilterInput {
+  """
+  A list of selectors for what routes/agencies should be allowed during search.
+    
+  In order to be excluded a route/agency has to match with at least one selector.
+    
+  An empty list or a list containing `null` is forbidden.
+  """
+  exclude: [TransitFilterSelectInput!]
+  """
+  A list of selectors for what routes/agencies should be excluded during the search.
+    
+  If route/agency matches with at least one selector it will be included.
+    
+  An empty list or a list containing `null` is forbidden.
+  """
+  include: [TransitFilterSelectInput!]
+}
+
+"""
+A list of selectors for filter including or excluding entities from the routing results. Null
+means everything is allowed to be returned and empty lists a not allowed.
+
+An entity will match with selectors if it matches with all non-empty lists.
+
+The `include` is always applied first, then `exclude`. If only `exclude` is present, the exclude is
+applied to the existing set of results.
```

Comment:
Feels like this should be on the upper type, not on this. Also, to me this wording is a bit vague. I think the "first" could be misinterpreted as a priority.